var/spack/repos/builtin/packages/py-flake8/package.py

```diff
@@ -25,13 +25,21 @@ class PyFlake8(PythonPackage):
     # However, py-flake8 requires py-setuptools during runtime as well.
     depends_on('py-setuptools@30:', type=('build', 'run'))
 
+    # entrypoints >= 0.3.0, < 0.4.0
+    # FIXME @0.3.0:0.3.999 causes concretization to hang
+    depends_on('py-entrypoints@0.3', when='@3.7.7', type=('build', 'run'))
+
+    # pyflakes >= 2.1.0, < 2.2.0
+    depends_on('py-pyflakes@2.1.0:2.1.999', when='@3.7.7', type=('build', 'run'))
     # pyflakes >= 1.5.0, < 1.7.0
     depends_on('py-pyflakes@1.5.0:1.6.999', when='@3.5.0', type=('build', 'run'))
     # pyflakes >= 0.8.1, != 1.2.0, != 1.2.1, != 1.2.2, < 1.3.0
     depends_on('py-pyflakes@0.8.1:1.1.0,1.2.3:1.2.3', when='@3.0.4', type=('build', 'run'))
     # pyflakes >= 0.8.1, < 1.1
     depends_on('py-pyflakes@0.8.1:1.0.0', when='@2.5.4', type=('build', 'run'))
 
+    # pycodestyle >= 2.5.0, < 2.6.0
+    depends_on('py-pycodestyle@2.5.0:2.5.999', when='@3.7.7', type=('build', 'run'))
```

Comment:
@michaelkuhn should this be `when='@3.7.7:'`? I assume all future `flake8` versions would need this

Reply:
Since the dependency version ranges for each flake8 vary a lot (see the other dependencies), a potential 3.8.x version will likely require some other version range for its dependencies. Therefore, I opted to follow the already existing style and limit this to @3.7.7.